### ISSUE TITLE
Broken code in the gallery / notebook execution flags not working

### ISSUE THREAD

**Issue by brian-rose** [ProjectPythia/xbatcher-ML-1-cookbook#11]
Currently we're publishing some broken code on our gallery, see here:
https://projectpythia.org/xbatcher-ML-1-cookbook/notebooks/surface_currents.html#data-and-preprocessing

This is a data access problem that's already been discussed in https://github.com/ProjectPythia/xbatcher-ML-1-cookbook/pull/4 and https://github.com/ProjectPythia/xbatcher-ML-1-cookbook/issues/5.

It seems like the workaround was to suppress notebook execution during the build. Currently we have in the `_config.yml`:
```
execute:
  # To execute notebooks via a binder instead, replace 'cache' with 'binder'
  execute_notebooks: force
  exclude_patterns:
    - "notebooks/surface_currents_prep.ipynb"
    - "notebooks/surface_currents_model.ipynb"
```

But those notebooks are executing anyway during the build. I'm not sure why those flags aren't working.

This repo is also missing the flag `nb_execution_raise_on_error: true` which would normally prevent our build system from publishing to GitHub Pages if there's a notebook error. PR https://github.com/ProjectPythia/xbatcher-ML-1-cookbook/pull/8 puts that flag in, but is currently failing the build (because the problematic notebook is still executing).

**Comment by brian-rose** [comment]
Ah, I see the issue. There is nothing wrong with exclude flags. However we are still executing the main notebook `notebooks/surface_currents.ipynb`, and that notebook itself calls the other two notebooks, e.g. with code like
```
%run ./surface_currents_prep.ipynb
```

I don't know if there's a real workaround here.

Maybe this cookbook should be removed from the gallery until the data access problem is resolved.

**Comment by brian-rose** [comment]
@mgrover1 just making you aware of this since you've been looking at this cookbook recently.

**Comment by cmdupuis3** [comment]
Just FYI I'm going to ask around again about the data hosting.

**Comment by brian-rose** [comment]
We are working on implementing a solution for cookbook data storage prior to our next [Hackathon](https://projectpythia.org/pythia-cookoff-2024/) this June.

**Comment by jukent** [comment]
I was trying to work on #12, but it's hard to parse or work on the error messages when i can't get the notebook to run at all locally. The first problem I encountered seems to be related to this issue?

When attempting to call `prepare_data(sc5, 200, 1000, 1000, 200)` I get the message 
```
ValueError: Bad Request: https://storage.googleapis.com/download/storage/v1/b/pangeo-cesm-pop/o/control%2F.zmetadata?alt=media
User project specified in the request is invalid.
```
(same as in [failing action](https://github.com/ProjectPythia/xbatcher-ML-1-cookbook/actions/runs/11770590976/job/32783238781#step:20:355)) which brings me to a blank page that says "Bucket is a requester pays bucket but no user project provided."
